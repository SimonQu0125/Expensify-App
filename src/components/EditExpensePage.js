import React from 'react';
import { connect} from 'react-redux';
import ExpenseForm from './ExpenseForm';
import { editExpense } from '../actions/expenses';
import { removeExpense } from '../actions/expenses';

export class EditExpensePage extends React.Component {
  onSubmit = (expense) => {
    this.props.editExpense(this.props.expense.id, expense);
    this.props.history.push('/');
  };

  onRemove = () => {
    this.props.removeExpense({ id: this.props.expense.id });
    this.props.history.push('/');
  };

  render(){
    return (
      <div>
        <ExpenseForm 
          //pass expense through props
          expense = {this.props.expense}
          //get the expense back by using child to call parent prop function
          onSubmit = {this.onSubmit}
        />
        <button onClick= {this.onRemove}>Remove</button>
      </div>
    );
  }

}

// const EditExpensePage = (props) => {
//   return (
//     <div>
//       <ExpenseForm 
//         //pass expense through props
//         expense = {props.expense}
//         //get the expense back by using child to call parent prop function
//         onSubmit = {(expense) => {
//           props.dispatch(editExpense(props.expense.id, expense));
//           props.history.push('/');
//         }}
//       />
//       <button onClick={() => {
//         props.dispatch(removeExpense({ id: props.expense.id }));
//         props.history.push('/');
//       }}>Remove</button>
//     </div>
//   );
// };

const mapStateToProps = (state,props) => {
  return {
    expense: state.expenses.find((e) => e.id === props.match.params.id )
  }
};

const mapDispatchToProps = (dispatch) => {
  return {
    editExpense: (id, expense) => dispatch(editExpense(id,expense)),
    removeExpense: (id) => dispatch(removeExpense(id))
  };
}

export default connect(mapStateToProps, mapDispatchToProps)(EditExpensePage);
