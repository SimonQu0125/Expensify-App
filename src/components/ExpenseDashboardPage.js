import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpenseSummary from '../components/ExpensesSummary';

const ExpenseDashboardPage = () => (
  <div>
    <ExpenseSummary />
    <ExpenseListFilters />
    <ExpenseList />
  </div>
);

export default ExpenseDashboardPage;
