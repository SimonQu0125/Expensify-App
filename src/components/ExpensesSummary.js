import React from 'react';
import { connect } from 'react-redux';
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/selectExpensesTotal';
import numeral from 'numeral'; 

export const ExpenseSummary = ({expensesCount, expensesTotal }) => (
  <div>
    <h1> 
      Viewing {expensesCount} {expensesCount === 1 ? 'expense':'expenses'} totalling {expensesTotal}. 
    </h1>
    
  </div>
);

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expensesCount: visibleExpenses.length,
    expensesTotal: numeral(selectExpensesTotal(visibleExpenses) / 100).format('$0,0.00')
  };
};

export default connect(mapStateToProps)(ExpenseSummary);