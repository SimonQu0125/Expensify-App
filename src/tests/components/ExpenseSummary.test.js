import React from 'react';
import { shallow } from 'enzyme';
import { ExpenseSummary } from '../../components/ExpensesSummary';

test('should correctly render page with 1 expense', () => {
  const wrapper = shallow(<ExpenseSummary expensesCount = {1} expensesTotal = {200} />);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render page with multiple expenses', () => {
  const wrapper = shallow(<ExpenseSummary expensesCount = {4} expensesTotal = {200212.12} />);
  expect(wrapper).toMatchSnapshot();
});