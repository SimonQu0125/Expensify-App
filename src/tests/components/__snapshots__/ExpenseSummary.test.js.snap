// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render page with 1 expense 1`] = `
<div>
  <h1>
    Viewing 
    1
     
    expense
     totalling 
    200
    .
  </h1>
</div>
`;

exports[`should correctly render page with multiple expenses 1`] = `
<div>
  <h1>
    Viewing 
    4
     
    expenses
     totalling 
    200212.12
    .
  </h1>
</div>
`;
